### PR TITLE
Fix bugs in Object Configuration Provider

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**

### CODE REVIEW

File: src/Okta.Sdk.Abstractions.UnitTests/CustomConfigurationProvidersShould.cs [hunk 1]
```diff
@@ -88,35 +88,117 @@ public void ParseEnvironmentConfiguration()
         [Fact]
         public void ParseConfigurationObject()
         {
-            var configurationObject = new
+            // anonymous type object
+            var anonymousFilled = new
```
Comment (bryanapellanes-okta) [inline src/Okta.Sdk.Abstractions.UnitTests/CustomConfigurationProvidersShould.cs:92]: I'm not clear what the intent of the suffix `Filled` is here, would `anonymousObject` communicate the intent more clearly?

Comment (bryanapellanes-okta) [reply src/Okta.Sdk.Abstractions.UnitTests/CustomConfigurationProvidersShould.cs:92]: Looking at the other variable names, perhaps `anonymousComplete` might communicate the intent for this object.

**Review by bryanapellanes-okta — APPROVED** [review]
Minor non-blocking naming comment.

File: src/Okta.Sdk.Abstractions.UnitTests/CustomConfigurationProvidersShould.cs (hunk 1 above)
Comment (bryanapellanes-okta) [reply src/Okta.Sdk.Abstractions.UnitTests/CustomConfigurationProvidersShould.cs:92]: Not a blocker, I just noticed there were other object names that included `partiallySet`; this implied that the `Filled` object is completely filled.  Mostly nitpicky semantics at this point.

Comment (laura-rodriguez) [reply src/Okta.Sdk.Abstractions.UnitTests/CustomConfigurationProvidersShould.cs:92]: I agree this name is not meaningful. What's the intent of this var?

Comment (andriizhegurov-okta) [reply src/Okta.Sdk.Abstractions.UnitTests/CustomConfigurationProvidersShould.cs:92]: This var is one of the cases I want to cover:  non-typed object (anonymous, not null) with some of fields set. 
I'll think about another name for this variable.

**Pull request merged**